### ISSUE TITLE
TypeError: Cannot read property 'getWrappedInstance' of undefined

### ISSUE THREAD

**Issue by Om1986** [airbnb/enzyme#1545]
Hi,

I am trying to access ChildComponent's methods in `TestDialog` component using `ref` and `getWrappedInstance()`. I couldn't avoid using `ref` as I had to re-use `ChildComponent` with and without `Modal` component and it's all methods.

 It worked fine in the browser but while writing unit tests for `TestDialog` component, I hit into error `TypeError: Cannot read property 'getWrappedInstance' of undefined` during `mount`  and executing `handleRequestSubmit`. Here's what I'm doing.

**TestDialog.js**
```
import ChildComponent from './ChildComponent';

class TestDialog () {
  constructor () {
    this. handleRequestSubmit = this.handleRequestSubmit.bind(this);
  }
  handleRequestSubmit() {
    const childRefContainer = this.childRefContainer.getWrappedInstance();
    // Do something with childRefContainer
  }

  render() {
    return (
      <Modal
        handleRequestSubmit={this.handleRequestSubmit}
        //some other props
      >
        <ChildComponent
          ref={(component) => this.childRefContainer = component}
      </Modal>
    );
  }
}

```

**ChildComponent.js**

```
import { translate } from 'react-i18next';
class ChildComponent() {

// render something
}
const ChildComponentWrapper = translate(['translation_key'], { withRef: true})(ChildComponent);
export default ChildComponentWrapper;

```
### Current behavior

Getting `TypeError: Cannot read property 'getWrappedInstance' of undefined` during `mount`  and executing `handleRequestSubmit`.

### Expected behavior

Able to test `handleRequestSubmit` method which is using  `getWrappedInstance`. 

### Your environment

OSX - 10.13.3
Node - 9.3.0
npm - 5.6.0

#### API

- [ ] shallow
- [x] mount
- [ ] render

#### Version

| library         | version
| ---------------- | -------
| Enzyme           |  3.3.0
| React            |  16.1.0

#### Adapter

- [x] enzyme-adapter-react-16
- [ ] enzyme-adapter-react-15
- [ ] enzyme-adapter-react-15.4
- [ ] enzyme-adapter-react-14
- [ ] enzyme-adapter-react-13
- [ ] enzyme-adapter-react-helper
- [ ] others ( )

**Comment by Zara603** [comment]
same problem, running jest

**Comment by ljharb** [comment]
I suspect this will be resolved by #1541 in the next release; will reopen if not.
